tests/testing-in-node.md

```diff
@@ -0,0 +1,127 @@
+# Testing In Node
+
+## Goal
+
+When writing any software, it is important to have confidence in the quality of the code. Tests can help you verify that your code is working as expected.
+
+We have looked at Jest and experimented with tests in CodeSandbox.  However, we have not yet used them in Node.js. In this section, we will learn how to use tests in Node.js and debugging tests with VS Code.
+
+Our goal for this lesson is to:
+
+- Practice using `yarn` to install dependencies
+- Apply our knowledge of Node.js to examine the `package.json` file
+- Use VS Code to examine and debug tests
+
+## Looking at a Sample Node.js Application
+
+Earlier in this lesson, we looked at a [CodeSandbox](https://codesandbox.io/s/fizzbuzz-with-tests-488sq?file=/test/fizzbuzz.test.js) application testing the `fizzBuzz` function. We will use the same project as a `Node.js` application.  
+
+To follow this lesson clone the [FizzBuzz-JavaScript](https://github.com/AdaGold/fizzbuzz_javascript) repository.
+
+### Install Dependencies
+
+Examine the `package.json` file.  Under the key `"devDependencies"` you can find a list of dependencies for the project.  Most of these simply involve getting the tests to run with the modern JavaScript features.
+
+- @babel/core
+- @babel/node
+- @babel/preset-env
+
+By default Node handles importing and exporting of JavaScript files differently, but using Babel we can use the modern JavaScript features.
+
+We have also included `jest` as a dependency.  This is our testing framework.
+
+To install these dependencies, run the following command in the project home directory:
+
+```bash
+$ yarn install
+```
+
+### Inspect the `test/fizzbuzz.test.js` file
+
+Open the `test/fizzbuzz.test.js` file.  This is a simple test file that tests the `fizzBuzz` function.  
+
+<details>
+  <summary>How does the test import the fizzBuzz function?</summary>
+
+  The test imports the `fizzBuzz` function with the statement import fizzBuzz from '../src/fizzbuzz';
+
+</details>
+
+### Running The Tests
+
+Under the `"scripts"` key in the `package.json` file we can find a command to run the tests.  This command is `"test"`.
+
+To run the tests in the terminal we can use following command:
+
+```bash
+$ yarn test
+```
+
+We should get the following output:
+
+```bash
```

Comment:
This shouldn't be highlighted as bash, it's just output.  (Trying to suggest changes here screws up the suggestion block though.  XD)